sklearn/utils/tests/test_utils.py

```diff
@@ -126,6 +130,28 @@ def test_pinvh_simple_complex():
     assert_almost_equal(np.dot(a, a_pinv), np.eye(3))
 
 
+def test_arpack_eigsh_initialization():
+    '''
+    Non-regression test that shows null-space computation is better with 
+    initialization of eigsh from [-1,1] instead of [0,1]
+    '''
+    random_state = check_random_state(42)
+
+    A = random_state.rand(50, 50)
+    A = np.dot(A.T, A)  # create s.p.d. matrix
+    A = graph_laplacian(A)
```

Comment:
This is strange. For me, the eigenvalue is mostly correct (~90%) with `uniform`, but it mostly fails (~99%) with `rand`.
Still, the algorithm applied to the Laplacian seems not stable enough to use it in the test.